addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -944,3 +974,36 @@ def mark_as_todo(self):
         self._compute_state()
 
         return
+
+
+def get_next_name(obj, name):
+    """
+    Get the next name for an object.
+
+    For when we want to create an object whose name links back to a previous
+    object.  For example BATCH/00001-02.
+    Assumes original names are of the form `r".*\d{5}"`.
+
+    Arguments:
+        obj - the source object for the name
+        name - the name for the object's model in the ir_sequence table
+
+    Returns:
+        The generated name, a string.
+    """
+    IrSequence = obj.env['ir.sequence']
+
+    # Name pattern for continuation object.
+    # Is two digits enough?
+    ir_sequence = IrSequence.search([('name', '=', name)], limit=1)
```

Comment:
It is better to search by code instead of name like `next_by_code()`

Reply:
Yes - I used `code` originally, but looking at `ir_sequence` code seems to be undefined for most models.  But happy to use it if you think that doesn't matter.

Reply:
The cases where we will end up using this function will be the ones with code. It would be good to do the same for `stock.picking` backorders.